docs/ContainerRuntimeConfigDesign.md

```diff
@@ -0,0 +1,95 @@
+# Summary
+
+Users need a way to update the container runtime configuration. Users can do this via MCO currently, but they need to know the correct options to use. We can hopefully help customers by documenting a CRD with the ContainerRuntime tuneables and have the MCO use these tuneables when rendering the crio.conf, storage.conf, and registries.conf files to Ignition/disk.
+
+The CRI-O runtime is used by default and is the recommended solution.
+
+# Motivation
+
+The MCO contains the logic to write the crio.conf, storage.conf, and registries.conf configuration files to Ignition. When Ignition starts on a machine it writes these three files to configure the container runtime. When these resources change within the MachineConfig, the new configs are rewritten and the MachineConfigDaemon is instructed to reboot the machine to pick up the new configs.
+
+## Goals
+
+The following goals should not require explicit knowledge of a customer to know the right command-line or configuration options to use within the ContainerRuntime.
+
+1. setting pids limit
+
+2. setting log level
```

Comment:
When would an end user set the log level (vs when we tell someone to set it during debugging)?

Reply:
This is more for admins enabling it for debugging.

Reply:
This is to me an issue for debugging , and wanting to canary a pool of nodes.

Reply:
If we have a canary pool the mco could easily select just those for debugging (but not sure we want this exposed to everyone like this risking to enable debug on every node).